src/mem_scan.c

```diff
@@ -77,11 +77,20 @@ void scan_mem(sqlite3 *db, char* arg_pid)
 	assert(psp != NULL);
 
 	kipp = procstat_getprocs(psp, KERN_PROC_PID, pid, &pcnt);
-	assert(kipp != NULL);
-	assert(pcnt == 1);
+	if (kipp == NULL) {
+		fprintf(stderr, "Unable to attach to process with pid %d, does it exist?\n", pid);
+		exit(1);
```

Comment:
These could potentially be `errx(1, "message", ...);`

Reply:
Thanks! Fixed.

Reply:
Note that `man 3 errx` says:
> The output is terminated by a newline character.

You can remove `\n` in your function calls.

Reply:
Thanks! Fixed and also added a new existence check in db_process for tables.